### DISCUSSION TITLE
shinyapps.io hard limit on package compilation time

### DISCUSSION THREAD

**Discussion by rfsaldanha** [duckdb/duckdb#5890] [General]
I had a problem trying to deploy a shiny app at shinyapps.io (from Posit) and got an answer from their premium support service. I thought it was worth registering it somewhere. 

After some unsuccessful tries with compilation time-out problems during deployment, I opened a ticket in the premium support service. Their response says that shinyapps.io has a hard limit for package compilation time of 15 minutes and `duckdb` needs more time on their servers.

> Hi,
>  
> Thank you for contacting Posit Support. 
> 
>  
> 
> Our investigation indicates that this issue is caused by the length of the build time for your application. We currently have a hard-coded limit of 15 minutes for building the deployment images. This max time is internal to shinyapps.io and is not user-configurable. In general, users do not run into this limit unless they have a large number of dependencies (in the hundreds) or are installing packages from source, requiring additional compilation time. Our suspicion is that your application is trying to install duckdb from source and the compilation time is exceeding the 15-minute limit.
>  
> 
> There are a couple of possible solutions:
> 
> We can open an issue to increase the maximum allowed time. This is a hard-coded limit in the server, but we could choose to increase it.  Note that this will require developer time and will not be available immediately.
> You can attempt to modify your code to use a different package than duckdb that doesn't need to be compiled.
>  
> 
> We understand that neither of these solutions provides a quick or easy fix for you, but you are currently running into a limitation of the shinyapps.io product.
> 
>  
> 
> Please let me know if you have any questions about this. Thank you for understanding.
> 
> Kind regards,

Here is the log of the deployment: https://pastes.io/a1ssqiv3ds

**Comment by hannes** [comment]
I will take this up with Posit directly, thanks for sharing.

**Comment by hannes** [comment]
Or maybe @krlmlr has an idea what we can do here?

**Comment by krlmlr** [comment]
This fell through the cracks.

Short answer: use binary packages.

Long answer: happy to expand if still relevant.
